### PR TITLE
sql parser support clickHouse

### LINKED ISSUE #31224: SQL parser support ClickHouse
## Feature Request

**For English only**, other languages will not be accepted.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Is your feature request related to a problem?

yes，issue[#28892](https://github.com/apache/shardingsphere/issues/28892)

### Describe the feature you would like.
I want to support Database ClickHouse，I will PR some code soon.

### PR DESCRIPTION
**Pull request by zzyReal666**
Fixes #31224.

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.
image

### CODE REVIEW

**Comment by zzyReal666** [comment]
> * The PR did not pass the code formatting validation. You can execute first `./mvnw spotless:apply -Pcheck -T1C`. Then execute `./mvnw checkstyle:check -Pcheck -T1C` to manually adjust to repair the CI.

Got it！

**Review by linghengqian — APPROVED** [review]
- LGTM. Just waiting for further review by other committers.

File: parser/sql/dialect/clickhouse/src/main/antlr4/imports/clickhouse/DMLStatement.g4 [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    : expr | DEFAULT | blobValue
+    ;
+
+blobValue
+    : STRING_
+    ;
+
+delete
+    :
+//    DELETE singleTableClause whereClause?
```
Comment (strongduanmu) [inline parser/sql/dialect/clickhouse/src/main/antlr4/imports/clickhouse/DMLStatement.g4:61]: Please remove useless sytax.

File: parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseDCLStatementVisitor.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+package org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.type;
+
+import org.apache.shardingsphere.sql.parser.api.visitor.statement.type.DCLStatementVisitor;
+import org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.ClickHouseStatementVisitor;
+
+/**
+ * ClickHouse DCL statement visitor.
+ */
+public class ClickHouseDCLStatementVisitor extends ClickHouseStatementVisitor implements DCLStatementVisitor {
```
Comment (strongduanmu) [inline parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseDCLStatementVisitor.java:26]: Please add final for this class.

File: parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseRLStatementVisitor.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+package org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.type;
+
+import org.apache.shardingsphere.sql.parser.api.visitor.statement.type.RLStatementVisitor;
+import org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.ClickHouseStatementVisitor;
+
+/**
+ * ClickHouse RL statement visitor.
+ */
+public class ClickHouseRLStatementVisitor extends ClickHouseStatementVisitor implements RLStatementVisitor {
```
Comment (strongduanmu) [inline parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseRLStatementVisitor.java:26]: Please add final for this class.

File: parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseTCLStatementVisitor.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+package org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.type;
+
+import org.apache.shardingsphere.sql.parser.api.visitor.statement.type.TCLStatementVisitor;
+import org.apache.shardingsphere.sql.parser.clickhouse.visitor.statement.ClickHouseStatementVisitor;
+
+/**
+ * ClickHouse TCL statement visitor.
+ */
+public class ClickHouseTCLStatementVisitor extends ClickHouseStatementVisitor implements TCLStatementVisitor {
```
Comment (strongduanmu) [inline parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseTCLStatementVisitor.java:26]: Please add final for this class.

Comment (zzyReal666) [reply parser/sql/dialect/clickhouse/src/main/java/org/apache/shardingsphere/sql/parser/clickhouse/visitor/statement/type/ClickHouseTCLStatementVisitor.java]: @strongduanmu  All done.

**Review by strongduanmu — APPROVED** [review]
Thank you for your contribution @zzyReal666

**Pull request merged**